From f4fd54b5c0719cc04b54c78374f849b2061b4fbc Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Fri, 31 May 2024 11:06:39 +0800
Subject: [PATCH 148/206] mtk: mt76: mt7996: move mt7996_scan_complete to
 main.c

The function mt7996_scan_complete() is moved to main.c so that following
commit can add calling to static faunctions in main.c

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 mt7996/mac.c  | 14 --------------
 mt7996/main.c | 14 ++++++++++++++
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/mt7996/mac.c b/mt7996/mac.c
index 7bf9199f..1113c535 100644
--- a/mt7996/mac.c
+++ b/mt7996/mac.c
@@ -2956,20 +2956,6 @@ unlock:
 	rcu_read_unlock();
 }
 
-void mt7996_scan_complete(struct mt7996_phy *phy, bool aborted)
-{
-	struct mt7996_vif *mvif = (struct mt7996_vif *)phy->scan_vif->drv_priv;
-	struct cfg80211_scan_info info = {
-		.aborted = aborted,
-	};
-
-	ieee80211_scan_completed(mvif->hw, &info);
-	phy->scan_chan = NULL;
-	phy->scan_req = NULL;
-	phy->scan_vif = NULL;
-	clear_bit(MT76_SCANNING, &phy->mt76->state);
-}
-
 static void mt7996_scan_check_sta(void *data, struct ieee80211_sta *sta)
 {
 	bool *has_sta = data;
diff --git a/mt7996/main.c b/mt7996/main.c
index fc9ff45d..2383c85e 100644
--- a/mt7996/main.c
+++ b/mt7996/main.c
@@ -2364,6 +2364,20 @@ mt7996_net_fill_forward_path(struct ieee80211_hw *hw,
 
 #endif
 
+void mt7996_scan_complete(struct mt7996_phy *phy, bool aborted)
+{
+	struct mt7996_vif *mvif = (struct mt7996_vif *)phy->scan_vif->drv_priv;
+	struct cfg80211_scan_info info = {
+		.aborted = aborted,
+	};
+
+	ieee80211_scan_completed(mvif->hw, &info);
+	phy->scan_chan = NULL;
+	phy->scan_req = NULL;
+	phy->scan_vif = NULL;
+	clear_bit(MT76_SCANNING, &phy->mt76->state);
+}
+
 static int
 mt7996_hw_scan(struct ieee80211_hw *hw, struct ieee80211_vif *vif,
 	       struct ieee80211_scan_request *hw_req)
-- 
2.18.0

